clients/src/main/resources/common/message/LeaderChangeMessage.json

```diff
@@ -22,7 +22,11 @@
     {"name": "LeaderId", "type": "int32", "versions": "0+",
       "about": "The ID of the newly elected leader"},
     {"name": "Voters", "type": "[]Voter", "versions": "0+",
-      "about": "The voters who voted for the current leader"}
+      "about": "The set of voters in the quorum for this epoch"},
+    {"name": "GrantingVoters", "type": "[]Voter", "versions": "0+",
+      "about": "The voters who voted for the leader at the time of election"},
+    {"name": "LeaderEpoch", "type": "int32", "versions": "0+",
```

Comment:
The reason we didn't have this is because the batch header already has a field for the leader epoch.

Reply:
hmm in that case, this might seem as redundant information then? I am sorry I have not seen the batch header message but is  that information would be handy when we view the LeaderChangeMessage? I felt adding it here might increase the debugging related information for Leadership changes and current state.

Reply:
To me the question that we need to answer is if we want this messages to be self contained. Meaning do we only need to have this message to know the state of the election or leader change. Or do we also need to know where in the log this message was inserted.

Reply:
@jsancio , from what I understood, looking at just the LeaderChangeMessage , the epoch won't be available. i looked at the code in `appendControlRecord` which gets invoked for a LeaderChangeMessage but i couldn't find it. The record appends to the log on it's own would have it. That's my understanding. 
With this information that I have, I think it makes sense to still add it unless it isn't agreed upon?

Reply:
hi @hachikuji , @jsancio , wanted to know if we plan to have the LeaderEpoch field in the LeaderChangeMessage.

Reply:
I think I'd rather leave it out. Conceptually it would be useful if it was included, but I am not sure what it actually buys us in practice and it introduces the possibility of inconsistency between the epoch in the message and that in the batch header.

Reply:
Thanks @hachikuji , I have removed it.